lib/attack.go

```diff
@@ -223,7 +223,61 @@ func Client(c *http.Client) func(*Attacker) {
 	return func(a *Attacker) { a.client = *c }
 }
 
-// A Rate of hits during an Attack.
+// A Rater defines the rate of hits during an Attack by
+// returning the duration an Attacker should wait until hiting
+// the next Target. When a negative value time is returned,
+// an Attacker will stop.
+type Rater interface {
+	Wait(began, now time.Time) time.Duration
+}
+
+// A RaterFunc is a function adapter type that implements the
+// Rater interface.
+type RaterFunc func(began, now time.Time) time.Duration
```

Comment:
>  Since we're expecting Rater implementors to track the hit count now, I would expect most implementations to require a struct or at least a uint64 for this anyway.

I believe with the `Pacer` design this isn't relevant anymore, but here's how you'd maintain state with a function closure.

```go
// NewLinearRater returns a Rater that increases it's wait time by
// the given duration every time it is called.
func NewLinearRater(step time.Duration) Rater {
    var count int64
    return RaterFunc(func(began, now time.Time) time.Duration {
        count++
        return began.Add(step * time.Duration(count))
    })
}
```